core/arch/aarch64/emit_utils.c

```diff
@@ -293,55 +294,70 @@ patch_stub(fragment_t *f, cache_pc stub_pc, cache_pc target_pc, cache_pc target_
     }
     /* target_pc is a far fragment. We must use an indirect branch. Note that the indirect
      * branch needs to be to the fragment prefix, as we need to restore the clobbered
-     * regs. */
+     * regs.
+     */
     byte *target_pc_slot = get_target_pc_slot(f, stub_pc);
     /* We set hot_patch to false as we are not modifying code. */
     ATOMIC_8BYTE_ALIGNED_WRITE(target_pc_slot, (ptr_uint_t)target_prefix_pc,
                                /*hot_patch=*/false);
     return;
 }
 
-bool
-stub_is_patched(dcontext_t *dcontext, fragment_t *f, cache_pc stub_pc)
+static bool
+stub_is_patched_for_intermediate_fragment_link(dcontext_t *dcontext, cache_pc stub_pc)
 {
     uint enc;
-    ATOMIC_4BYTE_ALIGNED_READ(vmcode_get_writable_addr(stub_pc), &enc);
-    /* linked to intermediate fragment. */
-    if ((enc & 0x14000000) == 0x14000000)
-        return true;
+    ATOMIC_4BYTE_ALIGNED_READ(stub_pc, &enc);
+    return (enc & 0x14000000) == 0x14000000;
```

Comment:
A named constant would help to know where this magic value comes from.

Reply:
I added a comment about the opcode, like many other spots in this file.